### PR TITLE
Ht 145 split tasks

### PR DESCRIPTION
**Pull request by feenster**
API for splitting a task

### CODE REVIEW

File: client/taskingmanager.config.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "development": {
     "EnvironmentConfig":{
-      "tmAPI": "http://tasking-manager-staging.eu-west-1.elasticbeanstalk.com/api/v1",
+      "tmAPI": "http://localhost:5000/api/v1",
```
Comment (hunt3ri) [inline client/taskingmanager.config.json]: Safe to check in?

File: server/__init__.py [hunk 2]
```diff
@@ -140,3 +141,4 @@ def init_flask_restful_routes(app):
     api.add_resource(UserSetLevel,                  '/api/v1/user/<string:username>/set-level/<string:level>')
     api.add_resource(UserAcceptLicense,             '/api/v1/user/accept-license/<int:license_id>')
     api.add_resource(IntersectingTilesAPI,          '/api/v1/grid/intersecting-tiles')
+    api.add_resource(SplitTaskAPI,                  '/api/v1/split/split-task/project/<int:project_id>/task/<int:task_id>')
```
Comment (hunt3ri) [inline server/__init__.py]: To make consistent with other APIs I'd change the shape to

'/api/v1/project/<int:project_id>/task/<int:task_id>/split

File: server/api/split_task_apis.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+from flask_restful import Resource, current_app, request
```
Comment (hunt3ri) [inline server/api/split_task_apis.py]: Unused ref to request

File: server/models/dtos/grid_dto.py [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 class GridDTO(Model):
     """ Describes JSON model used for creating grids """
     area_of_interest = BaseType(required=True, serialized_name='areaOfInterest')
     grid = BaseType(required=True)
-    clip_to_aoi = BooleanType(required=True, serialized_name='clipToAoi')
\ No newline at end of file
+    clip_to_aoi = BooleanType(required=True, serialized_name='clipToAoi')
+
+class SplitTaskDTO(Model):
+    """ DTO used to lock a task for mapping """
```
Comment (hunt3ri) [inline server/models/dtos/grid_dto.py]: Correct doc string

File: server/services/split_service.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+        # check its locked for mapping by the current user
+        if TaskStatus(original_task.task_status) != TaskStatus.LOCKED_FOR_MAPPING:
+            raise SplitServiceError('Status must be LOCKED_FOR_MAPPING to split')
+
+        if original_task.locked_by != split_task_dto.user_id:
+            raise SplitServiceError('Attempting to split a task owned by another user')
+
+        # check it's splittable
+        if not original_task.splittable:
```
Comment (hunt3ri) [inline server/services/split_service.py]: I'd move this to line 99, if task is not splittable there's no need to do any further checking.

File: client/taskingmanager.config.json (hunk 1 above)
Comment (feenster) [inline client/taskingmanager.config.json]: done

File: server/__init__.py (hunk 2 above)
Comment (feenster) [inline server/__init__.py]: done

File: server/api/split_task_apis.py (hunk 3 above)
Comment (feenster) [inline server/api/split_task_apis.py]: done

File: server/models/dtos/grid_dto.py (hunk 4 above)
Comment (feenster) [inline server/models/dtos/grid_dto.py]: done

File: server/services/split_service.py (hunk 5 above)
Comment (feenster) [inline server/services/split_service.py]: done

**Comment by feenster** [comment]
API for splitting a project task grid square

**Pull request merged**